cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/saved_queries_popover.spec.js

```diff
@@ -70,28 +78,90 @@ export const runSavedQueriesPopoverUITests = () => {
       cy.deleteIndex(INDEX_WITH_TIME_2);
     });
 
-    generateAllTestConfigurations(generateSavedTestConfiguration).forEach((config) => {
-      it(`should successfully create a saved query for ${config.testName}`, () => {
-        cy.navigateToWorkSpaceSpecificPage({
-          workspaceName,
-          page: 'discover',
-          isEnhancement: true,
+    const testConfigurations = generateAllTestConfigurations();
+
+    describe('should create initial saved queries', () => {
+      testConfigurations.forEach((config) => {
+        it(`should successfully create a saved query for ${config.testName}`, () => {
+          cy.navigateToWorkSpaceSpecificPage({
+            workspaceName,
+            page: 'discover',
+            isEnhancement: true,
+          });
+
+          cy.setDataset(config.dataset, datasourceName, config.datasetType);
+
+          cy.setQueryLanguage(config.language);
+          setDatePickerDatesAndSearchIfRelevant(config.language, START_TIME, END_TIME);
+
+          setQueryConfigurations(config);
+          verifyDiscoverPageState(config);
+          cy.saveQuery(config.saveName, ' ', true, true, false);
+        });
+      });
+    });
+
+    describe('should test loading, saving and deleting saved queries', () => {
+      testConfigurations.forEach((config) => {
+        it(`should load saved query: ${config.testName}`, () => {
+          cy.getElementByTestId('discoverNewButton').click();
+          setDatePickerDatesAndSearchIfRelevant(
+            config.language,
+            'Aug 29, 2020 @ 00:00:00.000',
+            'Aug 30, 2020 @ 00:00:00.000'
+          );
+
+          cy.loadSaveQuery(config.saveName, false);
+          // wait for saved queries to load.
+          cy.wait(2000);
+          verifyDiscoverPageState(config);
         });
 
-        cy.setDataset(config.dataset, datasourceName, config.datasetType);
+        it(`should update the loaded saved query: ${config.testName}`, () => {
+          updateAndVerifySavedQuery(config, false);
+        });
+
+        const saveAsNewQueryName = config.testName + SAVE_AS_NEW_QUERY_SUFFIX;
+        it(`should modify saved query: ${config.testName} and save as new query: ${saveAsNewQueryName}`, () => {
+          if (config.filters) {
+            cy.deleteAllFilters();
+          }
+          setDatePickerDatesAndSearchIfRelevant(config.language, START_TIME, END_TIME);
+
+          setQueryConfigurations(config);
+          verifyDiscoverPageState(config);
+          validateSaveAsNewQueryMatchingNameHasError(config.saveName, false);
+          cy.updateSaveQuery(saveAsNewQueryName, true, true, true, false);
+
+          cy.reload();
+          cy.getElementByTestId('discoverNewButton');
+          cy.loadSaveQuery(saveAsNewQueryName, false);
+          // wait for saved query to load
+          cy.wait(2000);
```

Comment:
Is there an element we can wait for instead of arbitrary wait to ensure that it has loaded? Perhaps the table w result?

In our various testing env this 2000 might not be sufficient, so its safer to wait for some element to be loaded